src/components/molecules/VenueJoinButton/VenueJoinButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback } from "react";
+import { VenueAccessMode } from "types/VenueAcccess";
+import { AnyVenue } from "types/venues";
+import { getCurrentTimeInUTCSeconds, getTimeBeforeParty } from "utils/time";
+import { joinVenue } from "utils/url";
+import { SecretPasswordForm } from "../SecretPasswordForm";
+
+interface VenueJoinButtonProps {
+  venueId?: string;
+  venue: AnyVenue;
+  onPasswordSubmit: () => void;
+  onPasswordSuccess: () => void;
```

Comment:
Make these optional so that callers don't need to potentially pass in noop functions + make sure to properly check they are defined before calling them later on.

Reply:
@denisdimitrov This still seems to be outstanding?

Reply:
Fixed.